src/Ion.ts

```diff
@@ -57,7 +58,7 @@ function makeBinaryReader(span: Span, options: Options) : BinaryReader {
 }
 
 function makeTextReader(span: Span, options: Options) : TextReader {
-  return new TextReader(span, options && options.catalog);
+  return new TextReader(span, options && options.catalog, options && options.raw_tokens);
```

Comment:
Agreed.

 As a means of keeping them  separate, lets add a new method `makeTokenReader` (or something like that) so that we can easily split clients calling the code.

The implementation, for now, can stay as is and we'll add an issue/todo for the parser/lexer split.